plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c

```diff
@@ -1105,7 +1105,19 @@ fu_logitech_bulkcontroller_device_write_firmware(FuDevice *device,
 	fu_progress_step_done(progress);
 
 	/* success! */
-	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+	if (fu_device_has_private_flag(device,
```

Comment:
would `if (FU_IS_LOGITECH_BULKCONTROLLER_CHILD(device)) {` work here?

Reply:
No. Pheripheral device is only visible/accessible to parent. It is not visible/accessible to OS/fwupd.  Only parent can update the firmware for this Pheripheral device

Reply:
Ahh, I see -- you're adding the flag to the proxy. What happens if you update the main device and the peripheral in the same transaction? I do wonder if we want a FuLogitechBulkcontrollerParent that inherits from FuLogitechBulkcontrollerDevice (like FuLogitechBulkcontrollerChild inherits from FuLogitechBulkcontrollerDevice) so that we can set the `FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG` in the parent `fu_logitech_bulkcontroller_parent_write_firmware()` function.

Reply:
Because each firmware image is ~2GB, we decided not to bundle multiple images together. With daisy chain of peripherals, it gets complicated. Based on D-Bus signal and leveraging CompositeId/ParentDeviceId, Compute host invokes fwupdmgr for each devices separately, in a pre-define order (i.e. update parent before child).